php-frontend/src/main/java/org/sonar/php/metrics/MetricsVisitor.java

```diff
@@ -0,0 +1,168 @@
+      functionComplexity += currentFunctionComplexity;
+      functionComplexityDistribution.add(currentFunctionComplexity);
+    }
+  }
+
+  @Override
+  public void leaveNode(Tree tree) {
+    if (tree.is(Kind.COMPILATION_UNIT)) {
+      saveComplexityMetrics(context());
+      saveCounterMetrics(context());
+      saveLineMetrics(context());
+    }
+  }
+
+  private void saveCounterMetrics(CheckContext context) {
+// FIXME UNCOMMENT TO ENABLE COUNTER VISITOR
+//    CounterVisitor counter = new CounterVisitor(context.tree());
+//    saveMetricOnFile(CoreMetrics.FUNCTIONS, counter.getFunctionNumber());
+//    saveMetricOnFile(CoreMetrics.STATEMENTS, counter.getStatementsNumber());
+//    saveMetricOnFile(CoreMetrics.ACCESSORS, counter.getAccessorsNumber());
+//    saveMetricOnFile(CoreMetrics.CLASSES, counter.getClassNumber());
+  }
+
+  private void saveComplexityMetrics(CheckContext context) {
+    int fileComplexity = NewComplexityVisitor.complexity(context.tree());
+
+    saveMetricOnFile(CoreMetrics.COMPLEXITY, fileComplexity);
+    saveMetricOnFile(CoreMetrics.COMPLEXITY_IN_CLASSES, classComplexity);
+    saveMetricOnFile(CoreMetrics.COMPLEXITY_IN_FUNCTIONS, functionComplexity);
+
+    sensorContext.saveMeasure(inputFile, functionComplexityDistribution.build(true).setPersistenceMode(PersistenceMode.MEMORY));
+
+    fileComplexityDistribution.add(fileComplexity);
+    sensorContext.saveMeasure(inputFile, fileComplexityDistribution.build().setPersistenceMode(PersistenceMode.MEMORY));
+  }
+
+  private void saveLineMetrics(CheckContext context) {
+// FIXME UNCOMMENT TO ENABLE LINE VISITOR
+// NOTE Should be migrated before comment metrics
+//    LineVisitor lineVisitor = new LineVisitor(context.tree());
+//    int linesNumber = lineVisitor.getLinesNumber();
+//    Set<Integer> linesOfCode = lineVisitor.getLinesOfCode();
+//
+//    saveMetricOnFile(CoreMetrics.NCLOC, lineVisitor.getLinesOfCodeNumber());
+//    saveMetricOnFile(CoreMetrics.LINES, linesNumber);
+
+// FIXME UNCOMMENT TO ENABLE COMMENT VISITOR
+    // fixme what about header comments?
+//    CommentLineVisitor commentVisitor = new CommentLineVisitor(context.tree(), ignoreHeaderComments);
+//    Set<Integer> commentLines = commentVisitor.getCommentLines();
+//
+//    saveMetricOnFile(CoreMetrics.COMMENT_LINES, commentVisitor.getCommentLineNumber());
+//    noSonarFilter.addComponent(sensorContext.getResource(inputFile).getEffectiveKey(), commentVisitor.noSonarLines());
+//
+//    FileLinesContext fileLinesContext = fileLinesContextFactory.createFor(inputFile);
+
+
+//    for (int line = 1; line <= linesNumber; line++) {
+      // FIXME UNCOMMENT TO ENABLE LINE VISITOR
+//      fileLinesContext.setIntValue(CoreMetrics.NCLOC_DATA_KEY, line, linesOfCode.contains(line) ? 1 : 0);
+
+    // FIXME UNCOMMENT TO ENABLE COMMENT VISITOR
+//    fileLinesContext.setIntValue(CoreMetrics.COMMENT_LINES_DATA_KEY, line, commentLines.contains(line) ? 1 : 0);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)